ambry-tools/src/integration-test/java/com.github.ambry/clustermap/HelixBootstrapUpgradeToolTest.java

```diff
@@ -87,24 +94,76 @@ public static void initialize() throws IOException {
+    }
+  }
+
+  @Parameterized.Parameters
+  public static List<Object[]> data() {
+    return Arrays.asList(new Object[][]{{"DC1"}, {"DC1, DC0"}, {"all"}});
+  }
+
   /**
    * Initialize ZKInfos for all dcs and start the ZK server.
    */
-  public HelixBootstrapUpgradeToolTest() throws Exception {
+  public HelixBootstrapUpgradeToolTest(String dcStr) {
     hardwareLayoutPath = tempDirPath + "/hardwareLayoutTest.json";
     partitionLayoutPath = tempDirPath + "/partitionLayoutTest.json";
     zkLayoutPath = tempDirPath + "/zkLayoutPath.json";
     zkJson = constructZkLayoutJSON(dcsToZkInfo.values());
     testHardwareLayout = constructInitialHardwareLayoutJSON(CLUSTER_NAME_IN_STATIC_CLUSTER_MAP);
     testPartitionLayout =
         constructInitialPartitionLayoutJSON(testHardwareLayout, DEFAULT_MAX_PARTITIONS_PER_RESOURCE, null);
+    this.dcStr = dcStr;
+    if (!dcStr.equalsIgnoreCase("all")) {
+      activeDcSet = Arrays.stream(dcStr.replaceAll("\\p{Space}", "").split(",")).collect(Collectors.toSet());
+    } else {
+      activeDcSet = new HashSet<>(Arrays.asList(dcs));
+    }
+  }
+
+  /**
+   * Test {@link HelixBootstrapUpgradeUtil#parseAndUpdateDcInfoFromArg(String, String)} method.
+   */
+  @Test
+  public void testParseDcSet() throws Exception {
+    assumeTrue(dcStr.equals("all"));
+    Utils.writeJsonObjectToFile(zkJson, zkLayoutPath);
+    Utils.writeJsonObjectToFile(testHardwareLayout.getHardwareLayout().toJSONObject(), hardwareLayoutPath);
+    Utils.writeJsonObjectToFile(testPartitionLayout.getPartitionLayout().toJSONObject(), partitionLayoutPath);
+    try {
+      HelixBootstrapUpgradeUtil.parseAndUpdateDcInfoFromArg(null, zkLayoutPath);
+      fail("Null dc string should fail");
+    } catch (IllegalArgumentException e) {
+    }
+    try {
+      HelixBootstrapUpgradeUtil.parseAndUpdateDcInfoFromArg("", zkLayoutPath);
+      fail("Empty dc string should fail");
+    } catch (IllegalArgumentException e) {
+    }
+    try {
+      HelixBootstrapUpgradeUtil.parseAndUpdateDcInfoFromArg("inv, inv1", zkLayoutPath);
+      fail("Invalid dc strings should fail");
+    } catch (IllegalArgumentException e) {
+    }
+    Set<String> expected = new HashSet<>(Collections.singletonList("DC1"));
+    Assert.assertEquals(expected, HelixBootstrapUpgradeUtil.parseAndUpdateDcInfoFromArg("DC1", zkLayoutPath).keySet());
+    expected.add("DC0");
+    Assert.assertEquals(expected,
+        HelixBootstrapUpgradeUtil.parseAndUpdateDcInfoFromArg("DC0, DC1", zkLayoutPath).keySet());
+    Assert.assertEquals(expected, HelixBootstrapUpgradeUtil.parseAndUpdateDcInfoFromArg("all", zkLayoutPath).keySet());
   }
 
   /**
    * Test the case where the zkHosts JSON does not have an entry for every Datacenter in the static clustermap.
    */
   @Test
   public void testIncompleteZKHostInfo() throws Exception {
+    assumeTrue(dcStr.equalsIgnoreCase("all") || dcStr.replaceAll("\\p{Space}", "").split(",").length == dcs.length);
```

Comment:
I think `assumeTrue` is unnecessary. Such test is supposed to be performed when all DCs are specified.

Reply:
The test is being performed when ALL (all dcs) is specified. `assumeTrue()` basically ensures that it is only run when this condition is true.
Basically, the test does not completely make sense when the input is not ALL. The static clustermap can contain more datacenters than what the user intends to update when input is not ALL, so this test is being skipped in those cases.